client/src/views/users/GreetUserModal.vue

```diff
@@ -274,8 +267,24 @@ function getNextButtonText(): string {
   return '';
 }
 
-function selectGuestSas(_code: string | null): void {
-  nextStep();
+async function selectGuestSas(code: string | null): Promise<void> {
+  if (!code) {
+    console.log('No code entered');
+    pageStep.value = GreetUserStep.WaitForGuest;
+    return;
+  }
+  if (code === greeter.value.correctSASCode) {
+    const result = await greeter.value.signifyTrust();
+    if (result.ok) {
+      await nextStep();
+    } else {
+      console.log('Signify trust failed', result.error);
+      pageStep.value = GreetUserStep.WaitForGuest;
+    }
+  } else {
+    console.log('Incorrect code entered');
+    pageStep.value = GreetUserStep.WaitForGuest;
+  }
```

Comment:
no abort on greeter side?

Reply:
Slightly updated to restart the process in this case, but this will have to be better handled during the complete tests.